src/components/contract-management/ContractManagementList.tsx

```diff
@@ -652,6 +662,106 @@ class ContractManagementList extends React.Component<Props, State> {
       contractsLoading: false
     });
   }
+
+  /**
+   * Method for opening file listing
+   */
+  private openFileListing = () => {
+    const input = this.xlsxInput.current;
+
+    if (!input) {
+      return;
+    }
+
+    input.click();
+  }
+
+  /**
+   * Method for importing xlsx file
+   *
+   * @param event event object
+   */
+  private importXlsx = async (event: React.ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.item(0);
+    event.target.value = "";
+
+    if (!file) {
+      return;
+    }
+
+    this.setState({ xlsxPreviewOpen: true });
+    this.setState({ parsedXlsxObjects: await this.parseXlsxFile(file) });
+  }
+
+  /**
+   * Method for parsing xlsx file
+   *
+   * @param file file
+   * @returns promise of conract preview data array
+   */
+  private parseXlsxFile = async (file: File): Promise<ContractPreviewData[]> => {
+    const { keycloak } = this.props;
+
+    if (!keycloak || !keycloak.token) {
+      return Promise.reject();
+    }
+
+    try {
+      const formData = new FormData();
+      formData.append("file", file);
+      const requestUrl = `${ process.env.REACT_APP_API_URL }/rest/v1/contractPreviews`;
+      const response = await fetch(requestUrl, {
+        method: "POST",
+        body: formData,
+        headers: {
+          "Authorization": `Bearer ${ keycloak.token }`
+        }
+      });
+      return await response.json();
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  /**
+   * Method for creating list of contracts to api
+   *
+   * @param contracts contract array
+   */
+  private createContracts = async (contracts: Contract[]) => {
+    const { keycloak } = this.props;
+
+    if (!keycloak || !keycloak.token) {
+      return;
+    }
+
+    try {
+      const contractsService = Api.getContractsService(keycloak.token);
+      const promises = contracts.map(contract =>
+        contractsService.createContract(contract)
+      );
+
+      const createdContracts = await Promise.all(promises);
+
+      this.setState({
+        xlsxPreviewOpen: false,
+        contracts: [...createdContracts, ...this.state.contracts]
+      });
+      await this.loadContacts(this.state.contracts);
+    } catch (error) {
+      console.log(`Couldn't create contracts: ${ error }`);
```

Comment:
no spaces between brackets in template literals
```suggestion
      console.log(`Couldn't create contracts: ${error}`);
```